eslint-plugin-expensify/tests/no-api-side-effect-actions.test.js

```diff
@@ -0,0 +1,86 @@
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../no-api-side-effect-actions');
+const message = require('../CONST').MESSAGE.NO_API_SIDE_EFFECTS_ACTIONS;
+
+const ruleTester = new RuleTester({
+    parserOptions: {
+        ecmaVersion: 6,
+        sourceType: 'module',
+    },
+});
+
+ruleTester.run('no-api-side-effect-actions', rule, {
+    valid: [
+        {
+            code: `
+                function test() {
+                    API.write('Report_AddComment', params);
+                }
+            `,
+            filename: './src/libs/actions/Test.js',
+        },
+        {
+            code: `
+                import Action from './src/libs/actions/Action.js'
+                function test() {
+                    Action.perform(params);
+                }
+            `,
+            filename: './src/libs/actions/Test.js',
+        },
+        {
+            // Test that two functions in one file with different API/action calls are valid
+            code: `
+                import Action from './src/libs/actions/Action.js'
+                function test() {
+                    API.write('Report_AddComment', params);
+                }
+                function test2() {
+                    Action.perform(params);
+                }
+            `,
+            filename: './src/libs/actions/Test.js',
+        },
+        {
+            // Make sure that we will not test a file which is not in '/actions/'
+            code: `
+                import Action from './src/libs/actions/Action.js'
+                function test() {
+                    API.write('Report_AddComment', params);
+                    Action.perform(params);
+                }
+            `,
+            filename: './src/libs/notActions/Test.js',
+        },
+    ],
+    invalid: [
+        {
+            code: `
+                import Action from './src/libs/actions/Action.js'
+                function test() {
+                    API.write('Report_AddComment', params);
+                    Action.perform(params);
+                }
+            `,
+            errors: [{
+                message,
+            }],
+            filename: './src/libs/actions/Test.js',
+        },
+        {
+            // Make sure chaining the API/Action calls is also invalid
+            code: `
+                import Action from './src/libs/actions/Action.js'
+                function test() {
+                    API.write('Report_AddComment', params).then((value) => {
+                        Action.perform(params);
+                    });
+                }
+            `,
+            errors: [{
+                message,
+            }],
+            filename: './src/libs/actions/Test.js',
```

Comment:
Invalid chaining test.

Reply:
Doesn't really make sense though as `API.write()` does not return a promise.

Reply:
Yeah, I think you would want to use `read` in this test case.

Reply:
Good points. I haven't really thought about how valid the examples are since the linter just goes through the tokens and searches for keywords.

I agree it will be better with real use example haha 😄